benchmark/operations/create_dataset.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2015 ClusterHQ Inc.  See LICENSE file for details.
+"""
+Operation to create a dataset.
+"""
+
+from functools import partial
+from uuid import uuid4
+
+from pyrsistent import PClass, field
+from zope.interface import implementer
+
+from flocker.common import loop_until
+
+from .._interfaces import IProbe, IOperation
+
+
+class EmptyClusterError(Exception):
+    """
+    Exception indicating that the cluster contains no nodes.
+    """
+
+
+@implementer(IProbe)
+class CreateDatasetConvergenceProbe(PClass):
+    """
+    Probe to create a dataset and wait for cluster to converge.
+    """
+
+    reactor = field(mandatory=True)
+    control_service = field(mandatory=True)
+    primary = field(mandatory=True)
+    dataset_id = field(mandatory=True)
+    volume_size = field(mandatory=True)
+
+    @classmethod
+    def setup(cls, reactor, control_service, dataset_id, volume_size):
+        """
+        Create a probe.
+
+        :param reactor: Twisted Reactor.
+        :param control_service: Benchmark control service.
+        :param UUID dataset_id: UUID for created dataset.
+        :param int volume_size: Size of created volume, in bytes.
+        :return: Deferred firing with a new probe.
+        """
+        d = control_service.list_nodes()
+
+        # Select an arbitrary node to be the primary for the dataset.
+        def pick_primary(nodes):
+            for node in nodes:
```

Comment:
`random.choice()` would be better, it's possible the order of nodes is determistic.